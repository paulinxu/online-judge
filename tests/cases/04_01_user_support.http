###
# Send request:
POST http://127.0.0.1:12345/jobs HTTP/1.1
content-type: application/json

{"contest_id":0,"language":"Rust","problem_id":0,"source_code":"fn main() { println!(\"Hello World!\"); }","user_id":1}
###
# Got response:
# HTTP 404
# content-length: 66
# content-type: application/json
# date: Thu, 25 Jul 2024 15:33:15 GMT
# 
# {"code":3,"message":"HTTP 404 Not Found","reason":"ERR_NOT_FOUND"}
###
# Send request:
GET http://127.0.0.1:12345/users HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 24
# content-type: application/json
# date: Thu, 25 Jul 2024 15:33:15 GMT
# 
# [{"id":0,"name":"root"}]
###
# Send request:
POST http://127.0.0.1:12345/users HTTP/1.1
content-type: application/json

{"name":"user1"}
###
# Got response:
# HTTP 200
# content-length: 23
# content-type: application/json
# date: Thu, 25 Jul 2024 15:33:15 GMT
# 
# {"id":1,"name":"user1"}
###
# Send request:
POST http://127.0.0.1:12345/jobs HTTP/1.1
content-type: application/json

{"contest_id":0,"language":"Rust","problem_id":0,"source_code":"fn main() { println!(\"Hello World!\"); }","user_id":1}
###
# Got response:
# HTTP 200
# content-length: 407
# content-type: application/json
# date: Thu, 25 Jul 2024 15:33:16 GMT
# 
# {"cases":[{"id":0,"info":"","result":"Compilation Success","time":0},{"id":1,"info":"","result":"Accepted","time":0}],"created_time":"2024-07-25T15:33:16.288452807Z","id":0,"result":"Accepted","score":100.0,"state":"Finished","submission":{"contest_id":0,"language":"Rust","problem_id":0,"source_code":"fn main() { println!(\"Hello World!\"); }","user_id":1},"updated_time":"2024-07-25T15:33:16.503954739Z"}
###
# Send request:
GET http://127.0.0.1:12345/users HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 48
# content-type: application/json
# date: Thu, 25 Jul 2024 15:33:16 GMT
# 
# [{"id":0,"name":"root"},{"id":1,"name":"user1"}]
###
# Send request:
POST http://127.0.0.1:12345/users HTTP/1.1
content-type: application/json

{"id":1,"name":"new_user1"}
###
# Got response:
# HTTP 200
# content-length: 27
# content-type: application/json
# date: Thu, 25 Jul 2024 15:33:16 GMT
# 
# {"id":1,"name":"new_user1"}
###
# Send request:
GET http://127.0.0.1:12345/users HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 52
# content-type: application/json
# date: Thu, 25 Jul 2024 15:33:16 GMT
# 
# [{"id":0,"name":"root"},{"id":1,"name":"new_user1"}]
###
# Send request:
POST http://127.0.0.1:12345/users HTTP/1.1
content-type: application/json

{"name":"new_user1"}
###
# Got response:
# HTTP 400
# content-length: 92
# content-type: application/json
# date: Thu, 25 Jul 2024 15:33:16 GMT
# 
# {"code":1,"message":"User name 'new_user1' already exists.","reason":"ERR_INVALID_ARGUMENT"}
###
# Send request:
POST http://127.0.0.1:12345/users HTTP/1.1
content-type: application/json

{"id":2,"name":"user2"}
###
# Got response:
# HTTP 404
# content-length: 65
# content-type: application/json
# date: Thu, 25 Jul 2024 15:33:16 GMT
# 
# {"code":3,"message":"User 2 not found.","reason":"ERR_NOT_FOUND"}
