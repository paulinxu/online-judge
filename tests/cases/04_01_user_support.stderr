[2024-07-25T15:33:15Z INFO  actix_server::builder] starting 16 workers
[2024-07-25T15:33:15Z INFO  actix_server::server] Actix runtime found; starting in Actix runtime
[2024-07-25T15:33:16Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET / HTTP/1.1" 404 0 "-" "-" 0.000261
[2024-07-25T15:33:16Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /jobs HTTP/1.1" 404 66 "-" "-" 0.000253
[2024-07-25T15:33:16Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /users HTTP/1.1" 200 24 "-" "-" 0.000068
[2024-07-25T15:33:16Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /users HTTP/1.1" 200 23 "-" "-" 0.000086
[2024-07-25T15:33:16Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /jobs HTTP/1.1" 200 407 "-" "-" 0.215736
[2024-07-25T15:33:16Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /users HTTP/1.1" 200 48 "-" "-" 0.000107
[2024-07-25T15:33:16Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /users HTTP/1.1" 200 27 "-" "-" 0.000121
[2024-07-25T15:33:16Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /users HTTP/1.1" 200 52 "-" "-" 0.000080
[2024-07-25T15:33:16Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /users HTTP/1.1" 400 92 "-" "-" 0.000095
[2024-07-25T15:33:16Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /users HTTP/1.1" 404 65 "-" "-" 0.000092
